RecoPixelVertexing/PixelTriplets/plugins/CAHitNtupletGeneratorKernels.cc

```diff
@@ -134,9 +134,9 @@ void CAHitNtupletGeneratorKernelsCPU::launchKernels(HitsOnCPU const &hh, TkSoA *
   // remove duplicates (tracks that share a doublet)
   kernel_earlyDuplicateRemover(device_theCells_.get(), device_nCells_, tracks_d, quality_d, params_.dupPassThrough_);
 
-  kernel_countMultiplicity(tuples_d, quality_d, device_tupleMultiplicity_.get());
+  kernel_countOrFillMultiplicity<CountOrFill::count>(tuples_d, quality_d, device_tupleMultiplicity_.get());
   cms::cuda::launchFinalize(device_tupleMultiplicity_.get(), cudaStream);
-  kernel_fillMultiplicity(tuples_d, quality_d, device_tupleMultiplicity_.get());
+  kernel_countOrFillMultiplicity<CountOrFill::fill>(tuples_d, quality_d, device_tupleMultiplicity_.get());
```

Comment:
what is the advantage of templating the kernel on `count` or `fill`, instead of having two kernels ?

Reply:
This is THE major code improvement.
The code of count and fill MUST be the same but for the call to either count (before the scan) or fill (once the size for each bucket has been determined).
Should be clearer if one look at the implementation